plonky2x/core/src/frontend/ecc/curve25519/curta/stark.rs

```diff
@@ -0,0 +1,348 @@
+use curta::chip::ec::edwards::ed25519::gadget::{CompressedPointGadget, CompressedPointWriter};
+use curta::chip::ec::edwards::ed25519::instruction::Ed25519FpInstruction;
+use curta::chip::ec::edwards::ed25519::point::CompressedPointRegister;
+use curta::chip::ec::gadget::EllipticCurveWriter;
+use curta::chip::ec::point::{AffinePoint, AffinePointRegister};
+use curta::chip::ec::scalar::ECScalarRegister;
+use curta::chip::register::Register;
+use curta::chip::trace::writer::{InnerWriterData, TraceWriter};
+use curta::chip::AirParameters;
+use curta::machine::builder::Builder;
+use curta::machine::ec::builder::EllipticCurveBuilder;
+use curta::machine::emulated::builder::EmulatedBuilder;
+use curta::machine::emulated::proof::EmulatedStarkProof;
+use curta::machine::emulated::stark::EmulatedStark;
+use curve25519_dalek::edwards::CompressedEdwardsY;
+use itertools::Itertools;
+use log::debug;
+use num_bigint::BigUint;
+use plonky2::iop::target::BoolTarget;
+use plonky2::util::log2_ceil;
+use plonky2::util::timing::TimingTree;
+
+use super::air_parameters::Ed25519AirParameters;
+use super::request::EcOpRequestType;
+use super::{Curve, ScalarField};
+use crate::frontend::curta::ec::point::{AffinePointVariable, CompressedEdwardsYVariable};
+use crate::frontend::curta::field::variable::FieldVariable;
+use crate::frontend::curta::proof::EmulatedStarkProofVariable;
+use crate::frontend::num::nonnative::nonnative::NonNativeVariable;
+use crate::prelude::*;
+
+pub enum Ed25519CurtaOp {
+    Add(
+        AffinePointRegister<Curve>,
+        AffinePointRegister<Curve>,
+        AffinePointRegister<Curve>,
+    ),
+    ScalarMul(
+        ECScalarRegister<Curve>,
+        AffinePointRegister<Curve>,
+        AffinePointRegister<Curve>,
+    ),
+    Decompress(CompressedPointRegister, AffinePointRegister<Curve>),
+    IsValid(AffinePointRegister<Curve>),
+}
+
+pub enum Ed25519OpVariable {
+    Add(
+        AffinePointVariable<Curve>,
+        AffinePointVariable<Curve>,
+        AffinePointVariable<Curve>,
+    ),
+    ScalarMul(
+        NonNativeVariable<ScalarField>,
+        AffinePointVariable<Curve>,
+        AffinePointVariable<Curve>,
+    ),
+    Decompress(Box<CompressedEdwardsYVariable>, AffinePointVariable<Curve>),
+    IsValid(AffinePointVariable<Curve>),
+}
+
+pub enum Ed25519CurtaOpValue {
+    Add(AffinePoint<Curve>, AffinePoint<Curve>, AffinePoint<Curve>),
+    ScalarMul(BigUint, AffinePoint<Curve>, AffinePoint<Curve>),
+    Decompress(CompressedEdwardsY, AffinePoint<Curve>),
+    IsValid(AffinePoint<Curve>),
+}
+
+pub struct Ed25519Stark<L: PlonkParameters<D>, const D: usize> {
+    stark: EmulatedStark<Ed25519AirParameters<L, D>, L::CurtaConfig, D>,
+    operations: Vec<Ed25519CurtaOp>,
+    degree: usize,
+}
+
+impl<L: PlonkParameters<D>, const D: usize> Ed25519Stark<L, D> {
+    pub fn new(request_data: &[EcOpRequestType]) -> Self {
+        let mut builder = EmulatedBuilder::<Ed25519AirParameters<L, D>>::new();
```

Comment:
I think adding a comment about the design of the stark will be helpful.  E.g. that all of ops other than ScalarMul will be public inputs and constrained similar to a snark and that the batch scalar mul op will be done via a stark.